src/common/src/buffer/bitmap.rs

```diff
@@ -191,12 +207,17 @@ pub struct Bitmap {
 
     /// Bits are stored in a compact form.
     /// They are packed into `usize`s.
-    bits: Box<[usize]>,
+    ///
+    /// Optimization: If all bits are set to 0 or 1, this field is `None`.
+    bits: Option<Box<[usize]>>,
 }
 
 impl EstimateSize for Bitmap {
     fn estimated_heap_size(&self) -> usize {
-        self.bits.len() * size_of::<usize>()
+        match &self.bits {
+            Some(bits) => bits.len() * size_of::<usize>(),
```

Comment:
Why don't we use https://doc.rust-lang.org/std/mem/fn.size_of_val.html or something similar?

Seems error prone to estimate size manually.